### ISSUE TITLE
Repeated object__compiled type in HPDL output

### ISSUE THREAD

**Issue by Maumagnaguagno** [panda-planner-dev/pandaPIparser#32]
Sometimes ``object__compiled`` is defined more than once in ``:types`` of generated HPDL, even for originally untyped domains.
Siadex generates a warning due to this.
The example below is one possible way to trigger such behavior, with objects and constants:
```Lisp
(define (domain test2)
  (:requirements :hierarchy)
  (:predicates (a ?v) (b))
  (:constants z)
  (:action act
    :parameters ()
    :precondition (a z)
    :effect (b)
  )
)
```
```Lisp
(define(problem repeatobject)
  (:domain test2)
  (:objects x y)
  (:htn :ordered-tasks (act))
  (:init (a z))
)
```
Which results in the following output:
```Lisp
(define (domain dom)
  (:requirements 
    :typing
    :htn-expansion
    :negative-preconditions
    :conditional-effects
    :universal-preconditions
    :disjunctive-preconditions
    :equality
    :existential-preconditions
  )
  (:types 
    object__compiled
    object__compiled
  )

  (:constants
    z - object__compiled
  )

  (:predicates
    (type_member_object__compiled ?var - object)
    (a ?var0 - object__compiled)
    (b)
  )


  (:task act
    :parameters ()
    (:method method1
      :precondition (
      )
      :tasks (
        (act_primitive)
      )
    )
  )

; ************************************************************
; ************************************************************
  (:action act_primitive
    :parameters ()
    :precondition (and
      (a z)
    )
    :effect (and
      (b)
    )
  )

)
```
```Lisp
(define (problem prob) (:domain dom)
  (:objects
    x y - object__compiled
  )

  (:init
    (a z)
    (type_member_object__compiled x)
    (type_member_object__compiled y)
    (type_member_object__compiled z)
  )

  (:tasks-goal
      :tasks (
        (act)
      )
  )
)
```

On a separate note, I would like to know why not use ``object`` as the default type and insert ``type_member_<type name>`` predicates.

**Comment by galvusdamor** [comment]
Hi Mau!

Thanks for pointing out this issue - which actually not only affects the SIADEX compiler but the parsing process for all domains. I've just pushed a commit (f0f3e3d809d2eb05a7457cb52b30433ec0df261c) that should fix the issue.

As for the naming of `object`. We default to `object_compiled` as in SIADEX, the type `object` has a special meaning - it is always the root type of the type hierarchy. HDDL domain can (and will) declare the type `object` lower down in the type hierarchy usually to mean something line "a physical object". This type has to be renamed for SIADEX to accept it. Thus we did it by default for all types named `object`. I've added (commit b67d88f) a special rule for the case that there is only one type in the problem (or none and everything is an undeclared object). The type will then be called `object`.

Gregor
